model/channel_member.go

```diff
@@ -11,12 +11,13 @@ import (
 )
 
 const (
-	CHANNEL_NOTIFY_DEFAULT      = "default"
-	CHANNEL_NOTIFY_ALL          = "all"
-	CHANNEL_NOTIFY_MENTION      = "mention"
-	CHANNEL_NOTIFY_NONE         = "none"
-	CHANNEL_MARK_UNREAD_ALL     = "all"
-	CHANNEL_MARK_UNREAD_MENTION = "mention"
+	CHANNEL_NOTIFY_DEFAULT          = "default"
+	CHANNEL_NOTIFY_ALL              = "all"
+	CHANNEL_NOTIFY_MENTION          = "mention"
+	CHANNEL_NOTIFY_NONE             = "none"
+	CHANNEL_MARK_UNREAD_ALL         = "all"
+	CHANNEL_MARK_UNREAD_MENTION     = "mention"
+	IGNORE_CHANNEL_MENTIONS_DEFAULT = "ignore_channel_mentions_not_set"
```

Comment:
I don't think we needed a third state for `ignore_channel_mentions`.  It's only either `true` or `false` with the default of `false`.

Reply:
Actually we need to know if the user has changed this setting explicitely or not. When the setting has not been touched yet, it is set at its default `ignore_channel_mentions_not_set` otherwise it is `ignore_channel_mentions_on`or `ignore_channel_mentions_off`

Reply:
I think default/on/off makes sense here since it matches the other settings although I know some other places where the "default" state is just the value not existing. This is probably nitpicking, but I would however say that we could get rid of the `ignore_channel_mentions_` prefix from the constant value and also use `default` instead of `not_set` for consistency, so the three values would be `default`, `on`, or `off`.